src/yadism/esf/tmc.py

```diff
@@ -557,48 +535,73 @@ class ESFTMC_g1(EvaluatedStructureFunctionTMC):
 
     def __init__(self, SF, kinematics):
         super().__init__(SF, kinematics)
-        # Kinematic factor for ZM g1
-        self._factor_shifted = self.x**2 / (self.xi * self.rho**3)
+        # Kinematic factor for ZM g1. Must be divided by `2\xi` to undo def.
+        self._factor_shifted = self.x / (self.xi * self.rho**3) / 2 / self.xi
         # Kinematic factor for common for `k1` & `k2` integral
-        self._factor_k1_k2 = (4 * self.x**2 * self.mu) / self.rho**3
+        self._factor_k1_k2 = (4 * self.x**2 * self.mu) / self.rho**4
+
+        # Extra (1/2) appears when undoing the `2x` in the definition.
         # Kinematic factor specific for `k1` integral
-        self._factor_k1 = (self.x + self.xi) / self.xi
+        self._factor_k1 = (self.x + self.xi) / self.xi / 2
         # Kinematic factor specific for `k2` integral
-        self._factor_k2 = (self.rho**2 - 3) / (2 * self.rho)
+        self._factor_k2 = (self.rho**2 - 3) / (2 * self.rho) / 2
 
     def _get_result_approx(self):
-        # NOTE: The approximations are evaluated at the lower
-        # integral limit; ie by integrating the factors of `g1^0`
-        approx_k1 = 1 - self.xi
-        approx_k2 = self.xi - 1 - np.log(self.xi)
-        # collect g1 results
+        # Collect g1 result.
         g1out = self.sf.get_esf(self.sf.obs_name, self._shifted_kinematics).get_result()
-        # Combine the expressions
-        return g1out * (
-            self._factor_shifted
-            + self._factor_k1_k2
-            * (self._factor_k1 * approx_k1 + self._factor_k2 * approx_k2)
+
+        # NOTE: The approximations are evaluated at the lower integral
+        # limit; ie by integrating the factors of `g1^0` from xi to 1.
+        # (1/2) appears when undoing the `2x` in the definition.
+        approx_k1 = (1 - self.xi) / self.xi
+        approx_k2 = 1 / self.xi - 1 + np.log(self.xi)
+
+        # Combine the expressions and putting back `2\xi`
+        factors = (
+            2
+            * self.xi
+            * (
+                self._factor_shifted
+                + self._factor_k1_k2
+                * (self._factor_k1 * approx_k1 + self._factor_k2 * approx_k2)
+            )
         )
+        return factors * g1out
 
     def _get_result_exact(self):
-        # collect g1 results
+        # Collect g1 result.
         g1out = self.sf.get_esf(self.sf.obs_name, self._shifted_kinematics).get_result()
+
         # Call to the raw integrals
-        k1out = self._k1()
+        k1out = self._h2()
```

Comment:
do you want to add an explicit redirect? else the names are a bit confusing ...

Reply:
I can indeed make a copy of that function (although this was what I tried to avoid) to make it more explicit.

As for the normalization, as long as everything is consistent, in that we account for a factor of $1/2$ from the integrand, $1/(2 \xi)$ everywhere $g_1$ appear, and undo by $2 \xi$ in overall, then everything is correct. This indeed I cross-checked a couple of times, and I think is also verified by the comparison plots in the Appendix of the draft.

Reply:
a simple redirect should be sufficient `_h2 = _k1` no?

Reply:
(adding updated docstrings is a good idea - however, just to say, also that simple line `_k1 = _h2` would have worked :upside_down_face: )